### PR TITLE
fix - add new button on opportunity board company picker doesnt work

### LINKED ISSUE #8455: Relation picker no result and new record improvements
# Current Behavior

f9dc8a8a45d268784f6b0efcff5e0d501d58a6782ee53636d2ec22368a5caefa

1. We display an unnecessarily prominent "No result" message.
2. It's unclear which object will have the record created.

# Desired Behavior

### 1 - No Result

Hide the "No result" message when the search or input yields no results.

### 2a - When Nothing is Typed in the Search

Display "Create _Object-name-singular_."

CleanShot 2024-11-11 at 15 04 25

### 2b - When Something is Typed in the Search

Display "Create _Object-name-singular: Record name_."

CleanShot 2024-11-11 at 15 03 28

### PR DESCRIPTION
**Pull request by ehconitin**
Closes https://github.com/twentyhq/twenty/issues/8455

https://github.com/user-attachments/assets/fec269b9-abb4-47e4-811e-fa23b69ca040

### CODE REVIEW

**Comment by ehconitin** [comment]
It seems like the add new button is itself removed!
@Bonapara was it supposed to be removed or reworked?

**Comment by Bonapara** [comment]
@ehconitin what do you mean? Do you have the issue link so I can remember? Thanks!

**Comment by ehconitin** [comment]
@Bonapara https://github.com/twentyhq/twenty/issues/8455
Can you see the add new button? I cant

**Comment by Bonapara** [comment]
Ohh I see you're right - Here a a few other issues:

File: packages/twenty-front/src/modules/search/hooks/useFilteredSearchEntityQuery.ts [hunk 1]
```diff
@@ -60,6 +60,7 @@ export const useFilteredSearchEntityQuery = ({
       filter: notFilter,
       limit: limit ?? DEFAULT_SEARCH_REQUEST_LIMIT,
       searchInput: searchFilter,
+      fetchPolicy: 'cache-and-network',
```
[the remark this answers is not in the document]
Comment (lucasbordeau) [reply packages/twenty-front/src/modules/search/hooks/useFilteredSearchEntityQuery.ts:63]: This should work out of the box, we shouldn't need to add this but maybe the optimistic effect isn't working properly for this request. It it's too much time consuming, let's create an issue to solve that.

**Review by charlesBochet — APPROVED** [review]
Merging this one, I'm doing another pass separately

**Pull request merged**